src/main/java/org/spongepowered/api/service/permission/PermissionService.java

```diff
@@ -153,6 +158,67 @@
      */
     Subject getDefaults();
 
+    /**
+     * Get data for a role template for a specific plugin. If the plugin has not set any permissions
+     * for the given role template, an empty {@link Optional} will be returned. Whether role template information
+     * is persistent or transient is implementation-dependent, though the final choice should most likely be
+     * up to the user.
+     *
+     * @param plugin The plugin to query the role template for.
+     * @param roleTemplate The specific role template identifier. Any string may be used,
+     *                     but {@link PermissionDescription} contains some common suggestions.
+     * @return An optional possibly containing the subject data for the given role template
+     */
+    default Optional<? extends SubjectData> getRoleTemplate(PluginContainer plugin, String roleTemplate) {
+        Objects.requireNonNull(plugin, "plugin");
+
+        return getCollection(SUBJECTS_ROLE_TEMPLATE).flatMap(coll ->
+                coll.getSubject(plugin.getMetadata().getId() + ":"
+                        + Objects.requireNonNull(roleTemplate, "roleTemplate")))
+                .map(Subject::getTransientSubjectData);
+    }
+
+
+    /**
+     * Get the data contained in role templates for all plugins providing data at the given key
+     *
+     * @param roleTemplate The specific role template identifier. Any string may be used,
+     *                     but {@link PermissionDescription} contains some common suggestions.
+     * @return An immutable set of mappings from plugin to subject data holder.
+     */
+    default Set<Map.Entry<PluginContainer, ? extends SubjectData>> getRoleTemplates(String roleTemplate) {
+        Optional<? extends SubjectCollection> coll = getCollection(SUBJECTS_ROLE_TEMPLATE);
+        if (!coll.isPresent()) {
+            return Collections.emptySet();
+        }
+
+        Optional<? extends Subject> globalSubj
+                = coll.get().getSubject(Objects.requireNonNull(roleTemplate, "roleTemplate"));
+        if (!globalSubj.isPresent()) {
+            return Collections.emptySet();
+        }
+
+        return Collections.unmodifiableSet(globalSubj.get()
+                .getTransientSubjectData().getParents(SubjectData.GLOBAL_CONTEXT).stream()
+                .map(SubjectReference::resolve)
+                .map(CompletableFuture::join)
```

Comment:
If we need to use `join` in the impl, maybe this method should return a `CompletableFuture` instead?

Reply:
I originally wrote this a while ago, but I think my reasoning was that these subjects would most likely be cached because they only exist from having transient data registered -- and transient data is expected to only be stored in memory.